### DISCUSSION TITLE
S3Resource#getFilename() returns null

### DISCUSSION THREAD

**Discussion by vedgunjan** [awspring/spring-cloud-aws#688] [Q&A]
Hi, 
I migrated to version 3.0.0-RC1 and in a spring batch application. Inside this application I'm converting S3 Object to Spring io.Resource and passing it to FlatFileItemReader. 

As S3Resource#getFilename() is not implemented and returning null, batch initialization is failing when `compareTo()` api is called. 
As a result I had to revert back to 
```
implementation group: 'com.amazonaws', name: 'aws-java-sdk-core', version: '1.11.986'
implementation group: 'com.amazonaws', name: 'aws-java-sdk-s3', version: '1.11.986'
```

where  SimpleStorageResource#getFilename() is implemented.

```
public class SimpleStorageResource extends AbstractResource implements WritableResource {
<<snip>>
	@Override
	public String getFilename() throws IllegalStateException {
		return this.objectName;
	}
<<snip>>
}
```

Do we have a fix or workaround for this issue?

Plz let me know.

Thanks
Ved

**Comment by maciejwalkowiak** [comment]
Thanks @vedgunjan for catching this. I don't think there is a workaround at this stage. Would you like to contribute a fix?

**Reply by vedgunjan** [reply]
@maciejwalkowiak what is the process of making submitting a PR with the code changes? Can you point me to the page which has setup instructions

**Reply by maciejwalkowiak** [reply]
@vedgunjan Sure!

How to build: https://github.com/awspring/spring-cloud-aws#checking-out-and-building
How to contribute: https://github.com/awspring/spring-cloud-aws/blob/main/CONTRIBUTING.md

**Comment by vedgunjan** [comment]
Hi Maciej,

Thanks for your response! And yes I would love to contribute to it.
I was going over the IntelliJ setup yesterday for the same reason.

Let me know if you would like me to proceed.

Regards,
Ved
